lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -232,13 +232,15 @@ public void close() {
                     .setKillEvaluator(EvaluatorRuntimeProtocol.KillEvaluatorProto.newBuilder().build())
                     .build();
             sendEvaluatorControlMessage(evaluatorControlProto);
+            this.stateManager.setClosing();
+          } else {
+            this.stateManager.setKilled();
           }
-        } finally {
+        } catch (Exception e) {
```

Comment:
> we should change it to  CLOSING  instead of  KILLED  if there is no exception and evaluator is running.

Ok, then where does the state change from `Closing` to `Killed` ?